### PR TITLE
JVECTOR-8

### PR DESCRIPTION
**Pull request by jkni**
Bring over roundtrip OnDiskGraphIndex tests and associated helper utilities.

This attempts to replace C*-isms with the appropriate JVector concepts, including paring down the test to not test irrelevant/nonexistent infrastructure.

### CODE REVIEW

File: src/test/java/com/github/jbellis/jvector/TestUtil.java [hunk 1]
```diff
@@ -1,9 +1,225 @@
+/*
```
Comment (jkni) [inline src/test/java/com/github/jbellis/jvector/TestUtil.java:1]: This is a ton of ugly boiler-plate. I do not know the right way to combine our header with the custom header used by ASF projects. I'm looking into this now.

File: src/test/java/com/github/jbellis/jvector/graph/TestOnDiskGraphIndex.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (jbellis) [inline src/test/java/com/github/jbellis/jvector/graph/TestOnDiskGraphIndex.java]: Okay to just use the DS header here -- we added the ASF boilerplate in anticipation of upstreaming to C* but it hasn't been contributed yet

Comment (jbellis) [reply src/test/java/com/github/jbellis/jvector/graph/TestOnDiskGraphIndex.java]: as with the other file we can just use the DS header

File: src/test/java/com/github/jbellis/jvector/TestUtil.java [hunk 3]
```diff
@@ -1,9 +1,225 @@
...
+        return new DataOutputStream(new BufferedOutputStream(Files.newOutputStream(outputPath)));
+    }
+
+    /**
+     * Deletes target Path and its contents (if applicable). Never throws an exception. Only suitable for tests.
+     * Visits all levels of file tree and does not follow symlinks.
+     * Exceptions will terminate the walk and print errors on stderr.
+     * @param targetPath Path to delete
+     */
+    public static void quietlyDelete(Path targetPath) {
```
Comment (jbellis) [inline src/test/java/com/github/jbellis/jvector/TestUtil.java]: can we call it deleteQuietly ?

**Review by jbellis — APPROVED** [review]

File: src/test/java/com/github/jbellis/jvector/TestUtil.java (hunk 3 above)
Comment (jkni) [reply src/test/java/com/github/jbellis/jvector/TestUtil.java]: Yes, change made

**Pull request merged**